src/js/components/ExpandingTable.js

```diff
@@ -17,7 +17,7 @@ class ExpandingTable extends React.Component {
   }
 
   expandRow(row) {
-    let {expandedRows} = this.state;
+    let {expandedRows} = Object.assign({}, this.state);
```

Comment:
but let's say we do something like `expandedRows.hello = 10` i think we're still mutating state. `expandedRows` in both cases would be the same object. the way around this would be to do `Object.assign({}, this.state.expandedRows)`, unless i'm missing something